src/views/catalog/utils/WizardTemplateContext.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+import { V1Template } from '@kubevirt-ui/kubevirt-api/console';
+
+type WizardTemplateContextType = {
+  /** the template used for the wizard */
+  template?: V1Template;
+  /**
+   * updates the template. The provided object will be merged with the existing template object
+   * @param template V1Template
+   */
+  updateTemplate?: (template: V1Template) => void;
+};
+
+export const WizardTemplateContext = React.createContext<WizardTemplateContextType>({});
+
+export const WizardTemplateContextProvider = WizardTemplateContext.Provider;
+
+export const useWizardTemplate = (): WizardTemplateContextType => {
+  const [template, setTemplate] = React.useState<V1Template>(() => getSessionStorage('template'));
```

Comment:
sessionStorage my guess isnt reactive and its async. Have you tested to see if data really update and cause rerender from changes in 2 different components?